src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/HttpClient/SendAsync.cs

```diff
@@ -175,11 +172,13 @@ private static void TryProcessResponse(IAgent agent, HttpResponseMessage respons
                 var httpStatusCode = response.StatusCode;
                 externalSegmentData.SetHttpStatusCode((int)httpStatusCode);
 
-                var headers = response.Headers?.ToList();
-                if (headers == null)
+                // Everything after this is for CAT, so bail if we're not using it
+                if (!agent.Configuration.CrossApplicationTracingEnabled)
```

Comment:
Do we need to check that DT is also not enabled? I think both can be enabled at the same time

Reply:
DT doesn't rely on response headers. 🙂

Reply:
Yeah, my initial comment had the logic wrong so I changed it. 🧇 

The concern was that CrossApplicationTracingEnabled may be set to true in the actual configuration at the same time as DistributedTracing/8T. I looked at the Configuration class and it does look like it checks that the CrossApplicationTracer is actually initialized before returning true here.

Reply:
@JcolemanNR  good point, i think the right check would be the following, because the current behavior is that DT setting will trump CAT setting.

`if (agent.Configuration.DistributedTracingEnabled  ||  !agent.Configuration.CrossApplicationTracingEnabled)`